src/nets/e2e_asr_th.py

```diff
@@ -1956,29 +1914,36 @@ def __init__(self, etype, idim, elayers, eunits, eprojs, subsample, dropout, in_
 
         self.etype = etype
 
-    def forward(self, xs, ilens):
+    def forward(self, xs_pad, ilens):
         '''Encoder forward
 
-        :param xs:
-        :param ilens:
-        :return:
+        :param torch.Tensor xs_pad: batch of padded input sequences (B, Tmax, D)
+        :param torch.Tensor ilens: batch of lengths of input sequences (B)
+        :return: batch of hidden state sequences (B, Tmax, erojs)
+        :rtype: torch.Tensor
         '''
         if self.etype == 'blstm':
-            xs, ilens = self.enc1(xs, ilens)
+            xs_pad, ilens = self.enc1(xs_pad, ilens)
         elif self.etype == 'blstmp':
-            xs, ilens = self.enc1(xs, ilens)
+            xs_pad, ilens = self.enc1(xs_pad, ilens)
         elif self.etype == 'vggblstmp':
-            xs, ilens = self.enc1(xs, ilens)
-            xs, ilens = self.enc2(xs, ilens)
+            xs_pad, ilens = self.enc1(xs_pad, ilens)
+            xs_pad, ilens = self.enc2(xs_pad, ilens)
         elif self.etype == 'vggblstm':
-            xs, ilens = self.enc1(xs, ilens)
-            xs, ilens = self.enc2(xs, ilens)
+            xs_pad, ilens = self.enc1(xs_pad, ilens)
+            xs_pad, ilens = self.enc2(xs_pad, ilens)
         else:
             logging.error(
                 "Error: need to specify an appropriate encoder archtecture")
             sys.exit()
 
-        return xs, ilens
+        return xs_pad, ilens
+
+    def _get_vgg2l_odim(self, idim, in_channel=3, out_channel=128):
```

Comment:
The same function is used for pytorch and chainer. You can move this to asr_utils.py or whatever.

Reply:
Integrated.